File: 0252-meeting-rooms/0252-meeting-rooms.cpp
class Solution {
public:
    bool canAttendMeetings(vector<vector<int>>& intervals) {
        if (intervals.size() == 0) return true;
        sort(intervals.begin(), intervals.end());
        vector<int> prev = intervals[0];
        for (int i = 1; i < intervals.size(); ++i){
            if (prev[1] > intervals[i][0]) return false;
            prev = intervals[i];
        }
        return true;
    }
};

File: 0252-meeting-rooms/0252-meeting-rooms.py
class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        intervals.sort()
        current = [0, 0]
        for start, end in intervals:
            if current[1] > start:
                return False
            current = [start, end]
        
        return True